packages/tracking/src/Tracking.php

```diff
@@ -66,20 +87,25 @@ function record_user_event( $event_type, $data = array(), $user = null ) {
 	/**
 	 * Record an event in Tracks - this is the preferred way to record events from PHP.
 	 *
-	 * @param mixed  $identity username, user_id, or WP_user object
-	 * @param string $event_name The name of the event
-	 * @param array  $properties Custom properties to send with the event
-	 * @param int    $event_timestamp_millis The time in millis since 1970-01-01 00:00:00 when the event occurred
+	 * @param mixed  $user                   username, user_id, or WP_user object.
+	 * @param string $event_name             The name of the event.
+	 * @param array  $properties             Custom properties to send with the event.
+	 * @param int    $event_timestamp_millis The time in millis since 1970-01-01 00:00:00 when the event occurred.
 	 *
 	 * @return bool true for success | \WP_Error if the event pixel could not be fired
 	 */
-	function tracks_record_event( $user, $event_name, $properties = array(), $event_timestamp_millis = false ) {
+	public function tracks_record_event( $user, $event_name, $properties = array(), $event_timestamp_millis = false ) {
 
 		// We don't want to track user events during unit tests/CI runs.
 		if ( $user instanceof \WP_User && 'wptests_capabilities' === $user->cap_key ) {
 			return false;
 		}
 
+		// Don't track users who have opted out or not agreed to our TOS, or are not running an active Jetpack.
+		if ( ! \Jetpack::jetpack_tos_agreed() ) {
```

Comment:
Since `Jetpack` is not a package this means that his package is not usable outside of jetpack. 
Which kind of defeats the purpose of this a package.

Reply:
That's a good point. Isn't it already an issue with this already?
https://github.com/Automattic/jetpack/blob/8a14c13d207c4b3a9650df6f453f99579ed07835/packages/tracking/legacy/class.tracks-client.php#L60

Does this mean we should also refrain from relying on constants like `JETPACK__PLUGIN_FILE` or `JETPACK__VERSION`, that are not set anywhere but in Jetpack right now?

Reply:
To be honest I don't quite know. 
but I always assume that the `src` folder is there code that is supposed to be working also outside of jetpack and the `src` are classes that could be used outside of Jetpack. 

I think using a filter here instead would work well.

Reply:
@enejb Is this what you had in mind? #13746

Reply:
The constants are a different thing FWIW, we have the `constants` package and we can use it to access any constants inside packages without relying on the core plugin.